vars/mtaBuild.groovy

```diff
@@ -28,6 +28,21 @@ def call(Map parameters = [:]) {
                                       parameters, parameterKeys,
                                       stepConfiguration, stepConfigurationKeys, stepDefaults)
 
+        JAVA_HOME_CHECK : {
+
+            // in case JAVA_HOME is not set, but java is in the path we should not fail
+            // in order to be backward compatible. Before introducing that check here
+            // is worked also in case JAVA_HOME was not set, but java was in the path.
+            // toolValidate works only upon JAVA_HOME and fails in case it is not set.
+
+            def rc = sh script: 'which java' , returnStatus: true
+            if(script.JAVA_HOME || (!script.JAVA_HOME && rc != 0)) {
```

Comment:
I would directly fail here if (!script.JAVA_HOME && rc != 0) instead of passing an empty script.JAVA_HOME to toolValidate. This would be better readable/understandable.

Reply:
In refactor commits like this I prefer to apply only these changes in logic which are really required. With the approach here we fail in the very same way like before. I would like to propose to adjust this in a later commit.